<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    <link rel="canonical" href="https://opensciencegrid.github.io/user-school-2018/materials/day4/part2-ex1-data-needs/">
    <link rel="shortcut icon" href="../../../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Exercise 2.1 - OSG User School 2018</title>
    <link href="../../../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../../../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../../../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">
    <link href="../../../css/extra.css" rel="stylesheet">
    <link href="../../../css/codehilite.css" rel="stylesheet">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../../../js/jquery-3.2.1.min.js"></script>
    <script src="../../../js/bootstrap-3.3.7.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '../../..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "Understanding Data Requirements", url: "#_top", children: [
              {title: "Background", url: "#background" },
              {title: "Understanding blast", url: "#understanding-blast" },
              {title: "Adding up data needs", url: "#adding-up-data-needs" },
              {title: "Talk about this as a group!", url: "#talk-about-this-as-a-group" },
              {title: "Up next!", url: "#up-next" },
          ]},
        ];

    </script>
    <script src="../../../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <h1 id="understanding-data-requirements">Understanding Data Requirements<a class="headerlink" href="#understanding-data-requirements" title="Permanent link">&para;</a></h1>
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<p>This exercise's goal is to learn to think critically about an application's data needs, especially before submitting a large batch of jobs or using tools for delivering large data to jobs. In this exercise we will attempt to understand the input and output of the bioinformatics application <a href="http://blast.ncbi.nlm.nih.gov/">BLAST</a>, which you used yesterday in Exercise 1.2.</p>
<h3 id="setup">Setup<a class="headerlink" href="#setup" title="Permanent link">&para;</a></h3>
<ul>
<li>Make sure you are logged into <code>user-training.osgconnect.net</code>.</li>
<li>Navigate to your <strong>stash</strong> directory within the home directory, and create a directory for this exercise named thur-blast.</li>
</ul>
<h4 id="copy-the-blast-executables">Copy the blast executables:<a class="headerlink" href="#copy-the-blast-executables" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="gp">user@user-training $</span> &lt;strong&gt;wget http://proxy.chtc.wisc.edu/SQUID/osgschool17/ncbi-blast-2.6.0+-x64-linux.tar.gz&lt;/strong&gt;
<span class="gp">user@user-training $</span> &lt;strong&gt;tar -xzf ncbi-blast-2.6.0+-x64-linux.tar.gz&lt;/strong&gt;
</pre></div>


<h4 id="copy-the-input-files">Copy the Input Files<a class="headerlink" href="#copy-the-input-files" title="Permanent link">&para;</a></h4>
<p>To run BLAST, we need an input file and reference database. For this example, we'll use the "pdbaa" database, which contains sequences for the protein structure from the Protein Data Bank. For our input file, we'll use an abbreviated fasta file with mouse genome information.</p>
<ol>
<li>Download these files to your current directory: \</li>
</ol>
<div class="codehilite"><pre><span></span><span class="gp">user@user-training $</span> &lt;strong&gt;wget http://proxy.chtc.wisc.edu/SQUID/osgschool17/pdbaa.tar.gz&lt;/strong&gt;
<span class="gp">user@user-training $</span> &lt;strong&gt;wget http://proxy.chtc.wisc.edu/SQUID/osgschool17/mouse.fa&lt;/strong&gt;
</pre></div>


<ol>
<li>Untar the <code>pdbaa</code> database: \</li>
</ol>
<div class="codehilite"><pre><span></span><span class="gp">user@user-training $</span> &lt;strong&gt;tar -xzf pdbaa.tar.gz&lt;/strong&gt;
</pre></div>


<h2 id="understanding-blast">Understanding blast<a class="headerlink" href="#understanding-blast" title="Permanent link">&para;</a></h2>
<p>Remember that <code>blastx</code> is executed in a command like the following:</p>
<div class="codehilite"><pre><span></span><span class="gp">user@user-training $</span> &lt;strong&gt;blastx -db <span style="color:red">database_rootname</span> -query <span style="color:red">input_file</span> -out <span style="color:red">results_file</span>&lt;/strong&gt;
</pre></div>


<p>In the above, the <code>input_file</code> is a file containing a number of genetic sequences, and the database that these are compared against is made up of several files that begin with the same root name (we previously used the "pdbaa" database, whose files all begin with <code>pdbaa</code>). The output from this analysis will be printed to a results file that is also indicated in the command.</p>
<h2 id="adding-up-data-needs">Adding up data needs<a class="headerlink" href="#adding-up-data-needs" title="Permanent link">&para;</a></h2>
<p>Looking at the files from the <code>blastx</code> jobs you ran yesterday, add up the amount of data that was needed for running the job. (If you deleted any of the files from yesterday, just resubmit the job before proceeding.) Here are the commands that will be most useful to you:</p>
<p>How to see the size of a specific file: <pre class="screen"> user@host $ <strong>ls -lh <span style="color:red">file<span class="twiki-macro ENDCOLOR"></span></strong> </pre></p>
<p>How to see the size of all files in the current directory: <pre class="screen"> user@host $ <strong>ls -lh</strong> </pre></p>
<p>How to determine the total amount of data in the current directory: <pre class="screen"> user@host $ <strong>du -sh <span style="color:red">directory<span class="twiki-macro ENDCOLOR"></span></strong> </pre></p>
<h3 id="input-requirements">Input requirements<a class="headerlink" href="#input-requirements" title="Permanent link">&para;</a></h3>
<p>Looking at yesterday's exercise, total up the amount of data in all of the files necessary to run the <code>blastx</code> job (which will include the executable, itself). Write down this number. Also take note of how much total data in in the <code>pdbaa</code> directory.</p>
<h3 id="output-requirements">Output requirements<a class="headerlink" href="#output-requirements" title="Permanent link">&para;</a></h3>
<p>The output that we care about from <code>blastx</code> is saved in the file whose name is indicated after the <code>-out</code> argument to <code>blastx</code>. However, remember that HTCondor also creates the error, output, and log files, which you'll need to add up, too. Are there any other files? Total all of these together, as well.</p>
<h2 id="talk-about-this-as-a-group">Talk about this as a group!<a class="headerlink" href="#talk-about-this-as-a-group" title="Permanent link">&para;</a></h2>
<p>Once you have completed the above tasks, we'll talk about the totals as a group.</p>
<ul>
<li>How much disk space is required on the submit server for one blastx run with the input file you used before?</li>
<li>How <em>many</em> files are needed and created for each run?</li>
<li>Assuming that each file is read completely by BLAST, and since you know how long blastx runs (time it):<ul>
<li>At what rate are files read in?</li>
<li>How many MB/s?</li>
</ul>
</li>
<li>How much total disk space would be necessary on the submit server to run 10 jobs? (remember that some of the files will be shared by all 10 jobs, and will not be multiplied)</li>
</ul>
<h2 id="up-next">Up next!<a class="headerlink" href="#up-next" title="Permanent link">&para;</a></h2>
<p>Next you will create a HTCondor submit script to transfer the Blast input files in order to run Blast on a worker nodes.</p>
<p><a href="../Education.UserSchool17Thurs22HTCondorFT">Next Exercise</a></p>

  <br>
    

  <div class="row wm-article-nav-buttons" role="navigation" aria-label="navigation">
    
    <div class="wm-article-nav pull-right">
      <a href="../part2-ex2-file-transfer/" class="btn btn-xs btn-default pull-right">
        Next
        <i class="fa fa-chevron-right" aria-hidden="true"></i>
      </a>
      <a href="../part2-ex2-file-transfer/" class="btn btn-xs btn-link">
        Exercise 2.2
      </a>
    </div>
    
    <div class="wm-article-nav">
      <a href="../part1-ex4-singularity/" class="btn btn-xs btn-default pull-left">
        <i class="fa fa-chevron-left" aria-hidden="true"></i>
        Previous</a><a href="../part1-ex4-singularity/" class="btn btn-xs btn-link">
        Exercise 1.4
      </a>
    </div>
    
  </div>

    <br>
</div>

<footer class="col-md-12 wm-page-content">
  <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p>
</footer>

</body>
</html>